pkg/multiclusterhub/multiclusterhub.go

```diff
@@ -0,0 +1,219 @@
+package multiclusterhub
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	mchv1 "github.com/stolostron/multiclusterhub-operator/api/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// MultiClusterHubBuilder provides struct for the MultiClusterHub object containing connection to
+// the cluster and the MultiClusterHub definitions.
+type MultiClusterHubBuilder struct {
+	Definition *mchv1.MultiClusterHub
+	Object     *mchv1.MultiClusterHub
+	errorMsg   string
+	apiClient  goclient.Client
+}
+
+// NewMultiClusterHubBuilder creates a new instance of MultiClusterHubBuilder.
+func NewMultiClusterHubBuilder(apiClient *clients.Settings, name, namespace string) *MultiClusterHubBuilder {
+	glog.V(100).Infof(
+		`Initializing new MultiClusterHub structure with the following params: name: %s, namespace:`,
+		name, namespace)
+
+	if apiClient == nil {
+		glog.V(100).Infof("apiClient cannot be nil")
+
+		return nil
+	}
+
+	builder := MultiClusterHubBuilder{
+		apiClient: apiClient.Client,
+		Definition: &mchv1.MultiClusterHub{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the MultiClusterHub is empty")
+
+		return nil
+	}
+
+	if namespace == "" {
+		glog.V(100).Infof("The namespace of the MultiClusterHub is empty")
+
+		return nil
+	}
```

Comment:
Since the builder has already been defined, you can just update builder.errorMsg with the appropriate error and return the builder at the bottom of the function